lib/daru/plotting/dataframe.rb

```diff
@@ -89,6 +91,21 @@ def plot_with_category opts
         plot.show
       end
 
+      SHAPES = ['circle','triangle-up', 'diamond', 'square', 'triangle-down', 'cross']
+      def get_shape
+        return to_enum(:get_shape) unless block_given?
+        loop do
```

Comment:
No it doesn't. I use `#next` to iterate over the enumerator. For example
```ruby
>> a = [1, 2, 3].cycle
=> #<Enumerator: [1, 2, 3]:cycle>
>> a.next
=> 1
>> a.next
=> 2
>> a.next
=> 3
>> a.next
=> 1
>> a.next
=> 2

```
BTW this code have been changed and I've replaced it with `#cycle`.